python/ray/data/_internal/remote_fn.py

```diff
@@ -18,12 +18,12 @@ def cached_remote_fn(fn: Any, **ray_remote_args) -> Any:
     """
     if fn not in CACHED_FUNCTIONS:
         default_ray_remote_args = {
-            "retry_exceptions": True,
```

Comment:
This default was set two years ago by https://github.com/ray-project/ray/pull/18296. But, I think we should get rid of it at this point for a couple reasons:
1. It's obsolete. https://github.com/ray-project/ray/pull/38773 added more specific retries. 
2. It can lead to bad UX. For example, if a non-transient error occurs, Ray will repeatedly retry the task, and you won't realize there's an error until your program potentially times out (or it never does!). Ideally, your program would immediately fail with the non-transient error.

Reply:
When I was doing some large-scale workloads, I depend on this flag to retry failed tasks due to AWS throttling or other temporary network issues.
If we remove it, we should allow retrying per op. E.g., set it to true only for read/write.

Reply:
@raulchen what were the specific errors you ran into? Shouldn't throttling be handled by https://github.com/ray-project/ray/pull/38773?

Reply:
One example is  `AWS Error NETWORK_CONNECTION during CreateMultipartUpload operation: curlCode: 28, Timeout was reached`.
I guess it's because write is covered by that PR.

Reply:
Weird. We should be performing retries for writes:

https://github.com/ray-project/ray/blob/8af874e834f8ae9c2bbfa8a9c76d434781c2048c/python/ray/data/datasource/file_datasink.py#L148-L155

@raulchen did you run into this issue after https://github.com/ray-project/ray/pull/38773 was merged? If so, I can add `retry_exceptions` for reads and writes.

Reply:
Yes, I got that error with last week's nightly. 
After a second thought, I feel it's not a good idea to simply add `retry_exceptions` for reads and writes. 
One opposite example is if users pass in a list of parquet files for read, where the files have different schemas. In this case, the read tasks will retry indefinitely without printing any info. 
Maybe it's better to just add retry for IO-related code. I think #38773 already covered read. We'll need to also do something similar for writes.
It's ok to do that in a follow-up PR.

Reply:
@bveeramani created an issue here https://github.com/ray-project/ray/issues/41211